tgui/packages/tgui/interfaces/SeedFabricator.js

```diff
@@ -0,0 +1,139 @@
+              MALFUNCTION
+              <Icon name="exclamation-triangle" pl={1.5} />
+            </Blink>
+            CHECK WIRES
+          </Modal>
+        )}
+        <Section>
+          <Flex>
+            <Flex.Item bold pr={1}>
+              Dispense:
+            </Flex.Item>
+            <Flex.Item basis={6} grow>
+              <NumberInput
+                value={dispenseAmount}
+                format={value => value + pluralize(" seed", value)}
+                minValue={1}
+                maxValue={10}
+                onDrag={(e, value) => setDispenseAmount(value)}
+              />
+            </Flex.Item>
+            <Flex.Item grow={2}>
+              <ProgressBar
+                value={Math.max(0, maxSeed - seedCount)}
+                maxValue={maxSeed}
+                ranges={{
+                  yellow: [5, Infinity],
+                  bad: [-Infinity, 5],
+                }}
+              >
+                <Icon name="bolt" />
+              </ProgressBar>
+            </Flex.Item>
+          </Flex>
+        </Section>
+        <Section>
+          {!canVend && (
+            <Modal textAlign="center"
+              width={25}
+              height={5}
+              fontSize={2}
+              fontFamily="Courier"
+              color="yellow">
+              <Blink interval={500} time={500}>
+                <Icon name="bolt" pr={1.5} />
+              </Blink>
+              UNIT RECHARGING
+            </Modal>
+          )}
+          {categories.map((category, index) => (
+            <SeedCategory
+              key={category.name}
+              category={category}
+              dispenseAmount={dispenseAmount} />
+          ))}
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const SeedCategory = (props, context) => {
+  const { act } = useBackend(context);
+  const { category, dispenseAmount } = props;
+  const { name } = category;
+  const seeds = category.seeds;
+  if (!seeds) return false;
+  seeds.sort((a, b) => a.name.localeCompare(b.name));
+
+  return (
+    <Collapsible
+      title={name}>
+      <Box>
+        {seeds.map((seed, index) => (
+          <Box key={seed.name} as="span">
+            <Button width="155px" height="32px" px={0} m={0.25}
+              onClick={() => act('disp', { path: seed.path, amount: dispenseAmount })}>
+              <Flex direction="row" align="center">
+                <Flex.Item>
+                  <img
+                    src={`data:image/png;base64,${seed.img}`}
+                    style={{
+                      'vertical-align': 'middle',
+                      'horizontal-align': 'middle',
+                    }}
+                    height="32px"
+                    width="32px" />
+                </Flex.Item>
+                <Flex.Item
+                  overflow="hidden"
+                  style={{ 'text-overflow': 'ellipsis' }}>
+                  {seed.name}
+                </Flex.Item>
```

Comment:
If hiding overflow (with ellipsis or not), should include a tooltip (native `title` tooltip is probably fine, unsure if `Flex.Item` passes that down nicely) so that the user can see the full text if they want to (or some other method of seeing the full text).